tests/extension/oneapi_composite_device/aspects.cpp

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2024 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include <sycl/sycl.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include "../../common/get_cts_object.h"
+
+#include <algorithm>
+#include <type_traits>
+#include <vector>
+
+namespace composite_device::tests {
+
+TEST_CASE("Test for aspects of composite devices",
+          "[oneapi_composite_device]") {
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+  auto platform = sycl_cts::util::get_cts_object::platform();
+  auto platform_composite_devices = platform.ext_oneapi_get_composite_devices();
+  auto all_composite_devices =
+      sycl::ext::oneapi::experimental::get_composite_devices();
+
+  INFO(
```

Comment:
Should we also check that the devices returned by `platform::get_composite_devices` are present in the result of the free function `get_composite_devices`?

Reply:
Added this test case into "enumerating composite devices" in bcebe1ad87778bca975673cc67eafaf1a1c23c26